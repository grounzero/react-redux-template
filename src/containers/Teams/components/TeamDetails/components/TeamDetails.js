import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button, Header, Segment, Form } from 'semantic-ui-react';

class TeamDetails extends Component {
    constructor(props) {
        super(props);

        const { team } = props;
        this.state = { id: team ? team.id : 0, name: team ? team.name : '' };
    }

    componentWillReceiveProps({ team }) {
        if (team) {
            const { id, name } = team;
            this.setState({ id, name });
        }
    }

    handleChange = (e, { name, value }) => this.setState({ [name]: value })

    handleSubmit = e => {
        this.props.actions.saveTeam(this.state)
    }

    render() {
        const { name } = this.state

        return (
            <article>
                <Header as='h2' attached='top'>
                    {name}
                </Header>
                <Segment attached>
                    <Form onSubmit={this.handleSubmit}>
                        <Form.Group widths='equal'>
                            <Form.Input label='Name' name='name' placeholder='Name' value={name} onChange={this.handleChange} />
                        </Form.Group>
                        <Button primary type='submit'>Save</Button>
                    </Form>
                </Segment>
            </article>
        );
    }
}

TeamDetails.propTypes = {
    team: PropTypes.shape({
        name: PropTypes.string.isRequired
    }),
    actions: PropTypes.shape({
        saveTeam: PropTypes.func.isRequired
    }).isRequired,
};

export default TeamDetails;