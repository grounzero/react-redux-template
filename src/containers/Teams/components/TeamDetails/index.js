import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import * as thunks from '../../thunks';
import TeamDetails from './components/TeamDetails';

class TeamDetailsContainer extends Component {
    componentDidMount() {
        const { team, actions } = this.props;
        if (team === undefined) {
            actions.getTeams();
        }
    }

    render() {
        const { team, actions } = this.props;
        return (
            <TeamDetails team={team} actions={actions} />
        );
    }
}

const mapStateToProps = ({ teams }, { match }) => {
    return {
        team: teams.models.filter(team => team.id === parseInt(match.params.teamId, 10))[0]
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        actions: bindActionCreators(thunks, dispatch)
    };
};

TeamDetailsContainer.propTypes = {
    actions: PropTypes.object.isRequired,
    team: PropTypes.object
};

export default connect(mapStateToProps, mapDispatchToProps)(TeamDetailsContainer);