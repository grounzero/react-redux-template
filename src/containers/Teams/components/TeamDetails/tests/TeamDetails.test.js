import React from 'react';
import renderer from 'react-test-renderer';

import TeamDetails from '../components/TeamDetails';

const team = { id: 0, name: ''};
const actions = {
    saveTeam: jest.fn()
};

it('TeamDetails renders correctly', () => {
    const tree = renderer.create(
        <TeamDetails team={team} actions={actions} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});