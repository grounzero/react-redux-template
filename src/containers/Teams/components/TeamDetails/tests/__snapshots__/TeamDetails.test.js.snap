// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamDetails renders correctly 1`] = `
<article>
  <h2
    className="ui top attached header"
  >
    
  </h2>
  <div
    className="ui attached segment"
  >
    <form
      action={undefined}
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="equal width fields"
      >
        <div
          className="field"
        >
          <label
            htmlFor={undefined}
          >
            Name
          </label>
          <div
            className="ui input"
          >
            <input
              disabled={undefined}
              name="name"
              onChange={[Function]}
              placeholder="Name"
              required={undefined}
              tabIndex={undefined}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <button
        className="ui primary button"
        disabled={undefined}
        onClick={[Function]}
        tabIndex={undefined}
        type="submit"
      >
        Save
      </button>
    </form>
  </div>
</article>
`;
